### PR TITLE
Remove unstable/unspecced login types.

### LINKED ISSUE #12278: Remove deprecated `m.login.jwt` login type
#7675 deprecated the unspecced `m.login.jwt` login identifier, replacing it with `org.matrix.login.jwt`. This landed in Synapse 1.16.0.

We should remove the deprecated `m.login.jwt` identifier at this point.

### PR DESCRIPTION
**Pull request by clokep**
This removes the login types:

* `m.login.jwt`, which was never specced and has been deprecated since Synapse 1.16.0.
* `uk.half-shot.msc2778.login.application_service`, which was stabilized as part of the Matrix spec v1.2 release.

As far as I can tell the application services have updated to the newer version.

Fixes #12278.

### CODE REVIEW

**Comment by clokep** [comment]
CC @tulir and @Half-Shot -- pretty sure nothing needs to be done here, but wanted to let you know.

File: changelog.d/12597.removal [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Remove the unspecified `m.login.jwt` login type and the unstable `uk.half-shot.msc2778.login.application_service` from
```
Comment (erikjohnston) [inline changelog.d/12597.removal:1]: Do we want to shout a bit louder about the fact we're *actually* removing `JWT` support?

**Review by erikjohnston — APPROVED** [review]

**Comment by Half-Shot** [comment]
@clokep for avoidance of doubt I checked that we removed our dependency on the unstable spec a while ago.

File: changelog.d/12597.removal (hunk 1 above)
Comment (clokep) [reply changelog.d/12597.removal:1]: > Do we want to shout a bit louder about the fact we're _actually_ removing `JWT` support?

We're not, we're only removing the unspecced `m.login.jwt` type. There's still `org.matrix.login.jwt` type as documented at https://matrix-org.github.io/synapse/develop/jwt.html

If anyone is using this they should be able to `s/m.login.jwt/org.matrix.login.jwt/g` and it should work fine.

What do you think?

**Pull request merged**